sdk/src/test/java/io/customer/sdk/store/DeviceStoreTest.kt

```diff
@@ -54,4 +27,29 @@ internal class DeviceStoreTest {
         deviceStore.buildUserAgent() `should be equal to`
             "Customer.io Android Client/1.0.0-alpha.6 (Google Pixel 6; 30) User App/1.0"
     }
+
+    @Test
+    fun `verify device attributes are created correctly`() {
+        val resultDeviceAttributes = deviceStore.buildDeviceAttributes()
+        val expectedDeviceAttributes = mapOf(
+            "device_os" to 30,
+            "device_model" to "Pixel 6",
+            "app_version" to "1.0",
+            "cio_sdk_version" to "1.0.0-alpha.6",
+            "device_locale" to "en",
+            "push_subscribed" to true
+        )
+
+        deviceStore.deviceBrand `should be equal to` "Google"
+        deviceStore.deviceModel `should be equal to` "Pixel 6"
+        deviceStore.deviceManufacturer `should be equal to` "Google"
+        deviceStore.deviceOSVersion `should be equal to` 30
+        deviceStore.deviceLocale `should be equal to` "en"
+
+        resultDeviceAttributes.keys shouldBeEqualTo expectedDeviceAttributes.keys
+
+        expectedDeviceAttributes.keys.forEach { key ->
+            expectedDeviceAttributes[key] `should be equal to` resultDeviceAttributes[key]
+        }
```

Comment:
```suggestion
        resultDeviceAttributes shouldBeEqualTo expectedDeviceAttributes
```

Should this work? I like this approach because it makes the test function future-proof. Each time that you update `DeviceStore.buildDeviceAttributes()`, you need to modify this test. If you use `shouldBeEqualTo`, you wouldn't need to. 

If it doesn't work, I am assuming it's because `shouldBeEqualTo` doesn't support `Map<String, Any>`? Could we create own our `shouldBeEqualToMap` to provide the same logic as:

```kotlin
resultDeviceAttributes.keys shouldBeEqualTo expectedDeviceAttributes.keys

        expectedDeviceAttributes.keys.forEach { key ->
            expectedDeviceAttributes[key] `should be equal to` resultDeviceAttributes[key]
        }
```